src/storage/src/source/reclock.rs

```diff
@@ -320,7 +320,20 @@ impl ReclockFollower {
 
 impl ReclockFollowerInner {
     pub fn compact(&mut self, new_since: Antichain<Timestamp>) {
-        assert!(PartialOrder::less_equal(&self.since, &new_since));
+        if PartialOrder::less_equal(&new_since, &self.since) {
```

Comment:
I think you meant  `if !PartialOrder::less_equal(&self.since, &new_since) {` here?

Reply:
it's flipped from before, we return when `new_since <= since`. Isn't that right? But the logging message does need fixing... 🙈

Reply:
@petrosagg not sure you're seeing this ☝️